### PR TITLE
Microwaves now have a single wire

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Microwaves have a single wire accessible when open, the activation wire.
When cut, the microwave will no longer function, when pulsed, the microwave
will turn on.
add: Stabilized dark purple extracts now cook items in your hands, rather than dropping
the cooked item on the floor.
/:cl:

Previously microwaves just worked on their contents, now they keep a subset of contents called `ingredients`, because otherwise it would explode whenever you put a signaler inside.

Someone asked me to do it. It seemed like a neat idea. God knows
what horrible things people will do with this.

### CODE REVIEW

**Comment by improvedname** [comment]
bless you <3

**Comment by XDTM** [comment]
Now we just need to be able to attach cellphones to wires

**Comment by Anonmare** [comment]
>Attach signaller to wire
>Put plasma tank inside
>Set off the signaller when the chef is at the microwave
>???
>The emergency shuttle has been called

**Comment by nicbn** [comment]
It'd be neat if you added a power wire and a wire that toggles the microwave "bug" that prevents it from working. And two wires to electrify it permanently or for some time, of course.

And while we're at it, why not add a wire that makes it more potent, burning all the food and making bigger explosions?

**Comment by evsey9** [comment]
Can you attach a signaller to the wire so that it sends a signal when the microwave turns on?

**Comment by MACIEKBAKI** [comment]
1. Attach a signaller to a pay stand
2. Attach signaller with same frequency to microwave
3. Make people pay for microwaved food
4. ???
5. Profit

**Comment by coiax** [comment]
@evsey9 I don't think you can extract signals from wire datums like that.

**Comment by XDTM** [comment]
My PR that allowed that is on hiatus until i manage to test/finish it

File: code/modules/food_and_drinks/kitchen_machinery/microwave.dm [hunk 1]
```diff
@@ -213,6 +229,14 @@
 /obj/machinery/microwave/proc/cook()
 	if(stat & (NOPOWER|BROKEN))
 		return
+	if(operating || broken > 0 || panel_open || !anchored || dirty == 100)
+		return
+
+	if(disabled)
+		audible_message("[src] buzzes.")
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/kitchen_machinery/microwave.dm]:
```suggestion
		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, 0)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by JMoldy** [comment]
This sounds like fun.

> Previously microwaves just worked on their contents, now they keep a subset of contents called ingredients, because otherwise it would explode whenever you put a signaler inside.

Does this mean that it was microwaving the machinery parts that went into it every time?

**Comment by coiax** [comment]
No clue where the internal components of a machine are stored. Probably in nullspace.

**Pull request merged**